### PR TITLE
expose kademlia replication factor in cli

### PR DESCRIPTION
**Pull request by klbrvik**

### CODE REVIEW

File: client/cli/src/params/network_params.rs [hunk 1]
```diff
@@ -127,6 +127,10 @@ pub struct NetworkParams {
 	#[arg(long)]
 	pub kademlia_disjoint_query_paths: bool,
 
+	/// Kademlia replication factor.
```
Comment (MathisWellmann) [inline client/cli/src/params/network_params.rs:130]: Could you give a brief explanation in the comment as to what this does?

File: client/network/src/discovery.rs [hunk 2]
```diff
@@ -196,10 +203,17 @@ impl DiscoveryConfig {
...
 			kademlia_disjoint_query_paths,
 			kademlia_protocols,
+			kademlia_replication_factor,
 		} = self;
 
 		let kademlia = if !kademlia_protocols.is_empty() {
 			let mut config = KademliaConfig::default();
+
+			if let Some(replication_factor) = kademlia_replication_factor {
+				dbg!(&kademlia_replication_factor);
```
Comment (MathisWellmann) [inline client/network/src/discovery.rs]: I'd say kill it or make it a regular log level

**Review by MathisWellmann — CHANGES_REQUESTED** [review]
Just wanna know what the replication factor means precisely

File: client/network/src/discovery.rs (hunk 2 above)
Comment (klbrvik) [reply client/network/src/discovery.rs]: Killing it.

File: client/cli/src/params/network_params.rs (hunk 1 above)
Comment (klbrvik) [reply client/cli/src/params/network_params.rs:130]: You mean in CLI arg comment or here?

Replication factor governs to how many other peers must value be sent to when inserting value into DHT for insertion to be successful, it also governs how many peers are queried when key lookup is being done.

Comment (klbrvik) [reply client/cli/src/params/network_params.rs:130]: By default it's set to 20 which is the reason why AuthorityDiscovery failed to track peer Multiaddrs. For development and testing purposes we want to configure this value.

**Review by drahnr — APPROVED** [review]

**Pull request merged**